core/unit_test/TestGraph.hpp

```diff
@@ -524,6 +509,38 @@ TEST_F(TEST_CATEGORY_FIXTURE(graph), force_global_launch) {
       }));
 
   listen_tool_events(Config::DisableAll());
+}
+
+template <typename>
+struct TestGraphSizedFunctorLaunch : public ::testing::Test {};
+
+using TestGraphSizedFunctorLaunchTypes = ::testing::Types<
+#if defined(KOKKOS_ENABLE_CUDA)
+  std::integral_constant<size_t, Kokkos::Impl::CudaTraits::ConstantMemoryUseThreshold>,
+  std::integral_constant<size_t, Kokkos::Impl::CudaTraits::ConstantMemoryUsage>
+#elif defined(KOKKOS_ENABLE_HIP) && defined(KOKKOS_IMPL_HIP_NATIVE_GRAPH)
+  std::integral_constant<size_t, Kokkos::Impl::HIPTraits::ConstantMemoryUseThreshold>,
+  std::integral_constant<size_t, Kokkos::Impl::HIPTraits::ConstantMemoryUsage>
+#else
+  void
+#endif
+>;
+
+TYPED_TEST_SUITE(TestGraphSizedFunctorLaunch, TestGraphSizedFunctorLaunchTypes,);
+
+TYPED_TEST(TestGraphSizedFunctorLaunch, sized_functor_launch) {
+#if defined(KOKKOS_ENABLE_CUDA)
+  if constexpr (!std::is_same_v<TEST_EXECSPACE, Kokkos::Cuda>) {
+#elif defined(KOKKOS_ENABLE_HIP) && defined(KOKKOS_IMPL_HIP_NATIVE_GRAPH)
+  if constexpr (!std::is_same_v<TEST_EXECSPACE, Kokkos::HIP>) {
+#endif
+    GTEST_SKIP() << "This execution space does not support launch mechanisms.";
+
+#if defined(KOKKOS_ENABLE_CUDA) || \
+    (defined(KOKKOS_ENABLE_HIP) && defined(KOKKOS_IMPL_HIP_NATIVE_GRAPH))
+  }
+
+test_sized_functor_launch<TypeParam::value + 1>();
```

Comment:
I think we'd gain in clarity if we design the whole thing as follows:
```c++
template <size_t Size>
struct SizedFunctor
{
public:
  using view_t = Kokkos:::View<int, ...>;

  static constexpr size_t padding = Size - sizeof(view_t);

private:
  std::byte unused[padding];
};
```
That is, the ˋSizedFunctorˋ is templated over the size we want it to have, instead of passing it the padding.

And later you'd just do:
```c++
void my_test()
{
  constexpr size_t my_launch_mechanism_threshold = ...;

  using sized_functor_t = SizedFunctor<my_launch_mechanism_threshold + 1>;
}
```